### PR TITLE
xwayland: allow to disable xwayland via config option

### PR DESCRIPTION
**Pull request by vnepogodin**
Adds ability to disable xwayland without recompilation with the flag

### CODE REVIEW

**Comment by vaxerski** [comment]
so it just adds an option but that option aint doing nothin'?

**Comment by vnepogodin** [comment]
> so it just adds an option but that option aint doing nothin'?

By default yes nothing will happen, because by default xwayland is enabled.

If you add:
```
xwayland {
  enabled = false
}
```
Then it wont spawn xwayland.

**Comment by vaxerski** [comment]
based on the diff I'd like to contest that statement

**Comment by vnepogodin** [comment]
lol the other file hasnt been added

**Comment by vaxerski** [comment]
well, add it then maybe? :)

**Comment by vnepogodin** [comment]
> well, add it then maybe? :)

yeah already pushed. sry haha

File: src/managers/XWaylandManager.cpp [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
 CHyprXWaylandManager::CHyprXWaylandManager() {
 #ifndef NO_XWAYLAND
+    static auto* const XWAYLAND_ENABLED = &g_pConfigManager->getConfigValuePtr("xwayland:enabled")->intValue;
+    if (!*XWAYLAND_ENABLED) {
```
Comment (vaxerski) [inline src/managers/XWaylandManager.cpp:13]: style nit: no {} around short ifs

File: src/managers/XWaylandManager.cpp [hunk 2]
```diff
@@ -9,6 +9,10 @@
 
 CHyprXWaylandManager::CHyprXWaylandManager() {
 #ifndef NO_XWAYLAND
+    static auto* const XWAYLAND_ENABLED = &g_pConfigManager->getConfigValuePtr("xwayland:enabled")->intValue;
```
Comment (vaxerski) [inline src/managers/XWaylandManager.cpp:12]: style nit: no_snake

**Review by vaxerski — COMMENTED** [review]
outside of this, since xwayland runs in parallel, you'll need to add a function (probably in the config manager) to check the status of xwayland on config reloads.

if someone edits the option, we might want to kill / start xwayland

**Comment by vnepogodin** [comment]
> outside of this, since xwayland runs in parallel, you'll need to add a function (probably in the config manager) to check the status of xwayland on config reloads.
> 
> if someone edits the option, we might want to kill / start xwayland

well I'd rather restrict editing that option. It should be initiated just once in the lifetime of the program

**Comment by vaxerski** [comment]
this is absolutely wrong, as xwayland will naturally be shut down if no xwayland apps are open, anyways (lazy mode in hyprland is always on)

File: src/managers/XWaylandManager.cpp (hunk 2 above)
Comment (vnepogodin) [reply src/managers/XWaylandManager.cpp]: that's screaming snake case, which is even used above )

all capitals just doesn't exist. thats little offtopic for a future

File: src/managers/XWaylandManager.cpp (hunk 1 above)
Comment (vnepogodin) [reply src/managers/XWaylandManager.cpp]: alright

**Comment by vnepogodin** [comment]
> this is absolutely wrong, as xwayland will naturally be shut down if no xwayland apps are open, anyways (lazy mode in hyprland is always on)

the intention for it is to force everything to native wayland. in lazy mode it will still launch as x11 app if broken detection with happened for it to fallback to xorg, etc

File: src/managers/XWaylandManager.cpp (hunk 2 above)
Comment (vaxerski) [reply src/managers/XWaylandManager.cpp]: > all capitals just doesn't exist. thats little offtopic for a future

I'd like to contest that information

image

**Comment by vaxerski** [comment]
yes, but what I want is for people to be able to dynamically toggle xwayland on / off.

As in, if xwayland is enabled and there are xwayland apps open, and someone disables it, kill xwayland.

> force everything to native wayland.

that is done by `setenv` and `unsetenv` of `$DISPLAY`

File: src/managers/XWaylandManager.cpp (hunk 2 above)
Comment (vnepogodin) [reply src/managers/XWaylandManager.cpp]: > > all capitals just doesn't exist. thats little offtopic for a future
> 
> I'd like to contest that information
> 
> image

thanks for showing me terrible code)

**Comment by vnepogodin** [comment]
> yes, but what I want is for people to be able to dynamically toggle xwayland on / off.
> 
> As in, if xwayland is enabled and there are xwayland apps open, and someone disables it, kill xwayland.
> 
> > force everything to native wayland.
> 
> that is done by `setenv` and `unsetenv` of `$DISPLAY`

well yeah I can look into that

**Comment by h0tc0d3** [comment]
You don’t need any extra parameters, you can do this and xwayland won’t start.
```
env = WLR_XWAYLAND, /dev/null
```

**Pull request closed without merging**